pkg/build/webhook/generic/generic.go

```diff
@@ -65,8 +66,13 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		}
 
 		var data api.GenericWebHookEvent
+		body, err = yaml.ToJSON(body)
```

Comment:
@liggitt  ?  
```
func ToJSON(data []byte) ([]byte, error) {
	if hasJSONPrefix(data) {
		return data, nil
	}
	return yaml.YAMLToJSON(data)
}
```

Basically this code now says "if you tell us it's yaml or json, we will convert it to json and then process it as json. If it was originally json, then we will no-op the conversion"

where does the error come in?